xamarin/Xamarin.Forms
# [Android/iOS] RemainingItemsThreshold and Scrolled implementation for CollectionView

### Description of Change ###

Not sure if this is the perfect implementation, but here's my take on "load on demand" scrolling for `CollectionView` as well as scroll event notifications. Here's a [demo](https://1drv.ms/v/s!AjlbPgOcTyP2dP_tvMTJjru423E) on Android. Implementation for UWP and Tizen is left out for others.

![giphy (2)](https://user-images.githubusercontent.com/16855542/55303077-c9e24980-5409-11e9-91e4-62dcd57d8b04.gif)

**Note:** You should review [UICollectionViewDataSourcePrefetching](https://developer.apple.com/documentation/uikit/uicollectionviewdatasourceprefetching) that was made available with iOS 10 before merging this code. I briefly looked at it but could not figure out how `RemainingItemsThreshold` could be used with prefetching.

### Issues Resolved ### 


- fixes #5623
- fixes #5716 
- fixes #4654
- fixes #6812

### API Changes ###
Added:
- RemainingItemsThreshold
- RemainingItemsThresholdReached (event)
- RemainingItemsThresholdCommand
- Scrolled (event)
- ScrolledEventArgs

### Platforms Affected ### 


- Core/XAML (all platforms)
- iOS
- Android

### Testing Procedure ###
Added an automated test but could not get it to run due to workspace issues with the UI test projects. For example, despite setting ANDROID_HOME variable, the UI test framework still seems to be confused about the location of this variable. Please run the test and if it doesn't work, please modify it to make sure it does.

### PR Checklist ###
